tgui/packages/tgui/interfaces/AirlockController.tsx

```diff
@@ -0,0 +1,168 @@
+
+/** Displays the buttons atop the window to cycle the airlock */
+const AirLockButtons = (_, context) => {
+  const { act, data } = useBackend<AirlockControllerData>(context);
+  const { airlockState } = data;
+
+  return (
+    ((airlockState === 'pressurize' || airlockState === 'depressurize') && (
+      <Button icon="stop-circle" onClick={() => act('abort')}>
+        Abort
+      </Button>
+    ))
+    || (airlockState === 'closed' && (
+      <>
+        <Button icon="lock-open" onClick={() => act('cycleInterior')}>
+          Open Interior Airlock
+        </Button>
+        <Button icon="lock-open" onClick={() => act('cycleExterior')}>
+          Open Exterior Airlock
+        </Button>
+      </>
+    ))
+    || (airlockState === 'inopen' && (
+      <>
+        <Button icon="lock" onClick={() => act('cycleClosed')}>
+          Close Interior Airlock
+        </Button>
+        <Button icon="sync" onClick={() => act('cycleExterior')}>
+          Cycle to Exterior Airlock
+        </Button>
+      </>
+    ))
+    || (airlockState === 'outopen' && (
+      <>
+        <Button icon="lock" onClick={() => act('cycleClosed')}>
+          Close Exterior Airlock
+        </Button>
+        <Button icon="sync" onClick={() => act('cycleInterior')}>
+          Cycle to Interior Airlock
+        </Button>
+      </>
+    ))
+  );
+};
+
+/** Displays the current status as two text strings, depending on door state. */
+const getAirlockStatus = (airlockState) => {
+  let currentStatus: AirlockStatus;
+
+  switch (airlockState) {
+    case 'inopen':
+      currentStatus = {
+        primary: 'Interior Airlock Open',
+        color: 'good',
+      };
+      break;
+    case 'pressurize':
+      currentStatus = {
+        ...defaultStatus,
+        primary: 'Cycling to Interior Airlock',
+        icon: 'fan',
+        color: 'average',
+      };
+      break;
+    case 'closed':
+      currentStatus = {
+        ...defaultStatus,
+        primary: 'Inactive',
+        color: 'white',
+      };
+      break;
+    case 'depressurize':
+      currentStatus = {
+        ...defaultStatus,
+        primary: 'Cycling to Exterior Airlock',
+        icon: 'fan',
+        color: 'average',
+      };
+      break;
+    case 'outopen':
+      currentStatus = {
+        ...defaultStatus,
+        primary: 'Exterior Airlock Open',
+        icon: 'exclamation-triangle',
+        color: 'bad',
+      };
+      break;
+    default:
+      currentStatus = defaultStatus;
+  }
+
+  return currentStatus;
+};
+
+/** Displays the numeric pressure alongside an icon for the user */
+const PressureIndicator = (props, context) => {
+  const { data } = useBackend<AirlockControllerData>(context);
+  const { sensorPressure } = data;
+  const { currentStatus } = props;
+  const { icon, color } = currentStatus;
+  let spin = icon === 'fan' ? 1 : 0;
+
+  return (
+    <Box color={color}>
+      {sensorPressure} kPa{' '}
+      {currentStatus.icon && <Icon name={icon} spin={spin} />}
```

Comment:
Ты уже сделал деструкцию `currentStatus` в 159 строке и получил нужные тебе `icon`, нет никакой необходимости снова обращаться к этому объекту.
```suggestion
      {icon && <Icon name={icon} spin={spin} />}
```